### DISCUSSION TITLE
i3wm Split Window Disappearing

### DISCUSSION THREAD

**Discussion by thewhizzard** [ghostty-org/ghostty#2584] [Q&A] [answered]
[Video of Bug](https://github.com/user-attachments/assets/f102bd25-f9fc-4761-bbf8-6f89526647a1)
When switching between focused terminals in Ghostty window with split tabs, occasionally the left-most will disappear, as shown in the video above. When this happens, its contents seem to appear briefly beneath the focused terminal before disappearing. This has been occurring on i3, with and without a compositor running. It's inconsistent, but happens frequently enough to be very noticeable.

This occurs only when the Ghostty window is the only one tiled in the workspace, and only when it is tiled, not floating. I first thought it might be opacity-related, but it still happens with opacity for both focused and unfocused terminals set to 1. I don't have any other X11 WMs on hand against which to test, but I'm unable to replicate the issue on Sway or Hyprland.

**Comment by mitchellh** [comment]
Can you please share your config?

**Comment by thewhizzard** [comment]
[config.txt](https://github.com/user-attachments/files/17608491/config.txt)
Mostly just the defaults, but unfortunately that makes it quite large.

**Comment by mitchellh** [comment, marked as answer]
Thanks. I'm moving this into a Q&A discussion until we can pinpoint a reproduction that at least one other person is able to get. At the time of writing this comment this isn't a very actionable issue. I'm an i3 user for years and have never seen this (I also use splits heavily). So I think we need more info in order to figure out what to do.

If/when we determine this is a Ghostty issue we can workaround, I will convert this back to an issue.

**Reply by thewhizzard** [reply]
Sounds good. It's possible that it's an issue somewhere else on my system - all I did to replicate the issue for the recorded video was switch between the two splits - it happened with both the mouse (since `focus-follows-mouse` is set) and, as shown, keyboard shortcuts.
